src/shared/lib/url-sanitization.ts

```
@@ -0,0 +1,282 @@
+    url: string;
+    foregroundColor?: string;
+    backgroundColor?: string;
+  };
+  errors?: string[];
+  warnings?: string[];
+}
+
+export function validateQrCodeParams(options: QrCodeValidationOptions): QrCodeValidationResult {
+  const errors: string[] = [];
+  const warnings: string[] = [];
+  const sanitizedParams: any = {};
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)